app/models/sources/joyent.rb

```diff
@@ -26,6 +26,24 @@ def validate_config!
       return false
     end
 
+    # Connnect to Joyent and find all of the machines with that combination of
+    # user and datacenter that is provided with each source object source.
+    # Each machine found will be delete.
+    def purge_servers(captcha=nil)
+      if captcha.nil?
+        puts("What is 1 + 2 + 3 + 4? Put answer as a parameter.")
+      elsif captcha == 10
+        connection.servers.select do |s|
```

Comment:
We don't need to preserve performance data. The goal of this feature is to completely clear out all traces of the proxies from a provider (hence, `purge_servers`).

From my sample code, the second loop covers all proxies, not just active ones. It is meant to delete from the database all `Proxy` object corresponding to this `Source`.